src/common/libutil/popen2.c

```diff
@@ -106,6 +131,10 @@ struct popen2_child *popen2 (const char *path, char *const argv[])
         saved_errno = errno;
         goto error;
     }
+    if ((p->flags & POPEN2_CAPTURE_STDERR) && pipe (p->efd) < 0) {
+        saved_errno = errno;
+        goto error;
+    }
```

Comment:
Not intentionally, though all fds are closed before exec in the child process by 
```c
    if (fdwalk (popen2_child_close_fd, p)) {
```
so it could be redundant to also set close-on-exec. I'll add it for consistency though.